pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -494,6 +516,9 @@ func TestReconcile(t *testing.T) {
 				},
 			}
 			result, err := r.Reconcile(ctx, request)
+			if &result == nil {
+				t.Errorf("Test case: %s. Expected: non nil result error, got: nil", tc.node.Name)
+			}
```

Comment:
Since the return here is not a point, it's impossible to get a nil, this check isn't needed. What are you trying to guard against?

Reply:
Basically I was fixing the lint warning.
You are right, it is not needed here :+1: